docs/tutorials/csharp/bert-nlp-csharp-console-app.md

```diff
@@ -0,0 +1,376 @@
+
+- The `inputs` variable indicates what the input shape will be. You can either create a dummy input like below, or use a sample input from testing the model.
+
+- Set the `opset_version` to the highest and compatible version with the model. Learn more about the opset versions [here](https://onnxruntime.ai/docs/reference/compatibility.html#:~:text=ONNX%20Runtime%20supports%20all%20opsets%20from%20the%20latest,with%20ONNX%20opset%20versions%20in%20the%20range%20%5B7-9%5D.).
+
+- Set the `input_names` and `output_names` for the model.
+
+- Set the `dynamic_axes` for the dynamic length input because the `sentence` and `context` variables will be of different lengths for each question inferenced.
+
+```python
+# Generate dummy inputs to the model. Adjust if neccessary.
+inputs = {
+        # list of numerical ids for the tokenized text
+        'input_ids':   torch.randint(32, [1, 32], dtype=torch.long), 
+        # dummy list of ones
+        'attention_mask': torch.ones([1, 32], dtype=torch.long),     
+        # dummy list of ones
+        'token_type_ids':  torch.ones([1, 32], dtype=torch.long)     
+    }
+
+symbolic_names = {0: 'batch_size', 1: 'max_seq_len'}
+torch.onnx.export(model,                                         
+# model being run
+                  (inputs['input_ids'],
+                   inputs['attention_mask'], 
+                   inputs['token_type_ids']),                    # model input (or a tuple for multiple inputs)
+                  model_path,                                    # where to save the model (can be a file or file-like object)
+                  opset_version=11,                              # the ONNX version to export the model to
+                  do_constant_folding=True,                      # whether to execute constant folding for optimization
+                  input_names=['input_ids',
+                               'input_mask', 
+                               'segment_ids'],                   # the model's input names
+                  output_names=['start_logits', "end_logits"],   # the model's output names
+                  dynamic_axes={'input_ids': symbolic_names,
+                                'input_mask' : symbolic_names,
+                                'segment_ids' : symbolic_names,
+                                'start_logits' : symbolic_names, 
+                                'end_logits': symbolic_names})   # variable length axes/dynamic input
+```
+## Understanding the model in Python
+When taking a prebuilt model and operationalizing it, its useful to take a moment and understand the models pre and post processing, and the input/output shapes and labels. Many models have sample code provided in Python. We will be inferencing our model with C# but first lets test it and see how its done in Python. This will help us with our C# logic in the next step.
+
+- Create the `preprocess`, `postprocess`, `init` and `run` functions in python to test the model.
+
+```python
```

Comment:
Since this code is (mostly?) copied from https://onnxruntime.ai/docs/tutorials/azureml.html, I would link to it here rather than copy. It will help with the focus of your tutorial, which is C# and also do better with SEO (duplicated content tends to harm perf of both pages)

Reply:
I was wondering that! Updated!